src/components/Button/buttonContent.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { StyledText } from './styled';
+import { BaseProps, ButtonVariant } from '../types';
+
+interface Props extends BaseProps {
```

Comment:
If we will not use style prop here then there is no need to extends from BaseProps, then you can:
```
interface Props {
    variant?: ButtonVariant;
    disabled?: boolean;
    children?: any;
}
```